### PR TITLE
suricata-update: Initial commit

### PR DESCRIPTION
**Pull request by Grommish**
Provide the framework for Suricata-update

Signed-off-by: Donald Hoskins <grommish@gmail.com>

Maintainer: me 
Compile tested: (put here arch, model, OpenWrt version)
Run tested: (put here arch, model, OpenWrt version, tests done)

Description:

Suricata-update framework, uci config and init.d service entry.

Requires [Suricata6](https://github.com/openwrt/packages/pull/13924)

### CODE REVIEW

File: net/suricata-update/files/install_suricata_update [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh /etc/rc.common
+/usr/bin/python3 -m pip install --upgrade pip
```
Comment (neheb) [inline net/suricata-update/files/install_suricata_update:2]: uhhhh what?

**Comment by Grommish** [comment]
The whole thing is ludicrous, but pip provides suricata-update, but I need
to get the config and init.d files to the device.  I am still working on
trying to get Package/postinst to work.

So...  Suggestions?

File: net/suricata-update/files/install_suricata_update (hunk 1 above)
Comment (Grommish) [reply net/suricata-update/files/install_suricata_update:2]: It ensures pip3 is up to date, since it's currently outdated in master.  Not a great idea?

**Comment by neheb** [comment]
ping @jefferyto

**Comment by jefferyto** [comment]
There are lots of Python program packages in the repo, e.g. [youtube-dl](https://github.com/openwrt/packages/blob/master/multimedia/youtube-dl/Makefile), [docker-compose](https://github.com/openwrt/packages/blob/master/utils/docker-compose/Makefile), etc. If you search for "python3-package.mk" in the repo you will find many other examples. I suggest you take one of these makefiles as a template.

The Python packages folder readme also has a section on [building a Python package](https://github.com/openwrt/packages/blob/master/lang/python/README.md#building-a-python-package) which explains most (if not all) of the various Python package variables.

If you run into any problems writing a Python package makefile, there are lots of folks here who can help, but I'm afraid your `install_suricata_update` is definitely not the way to do this.

**Comment by Grommish** [comment]
> 
> 
> There are lots of Python program packages in the repo, e.g. [youtube-dl](https://github.com/openwrt/packages/blob/master/multimedia/youtube-dl/Makefile), [docker-compose](https://github.com/openwrt/packages/blob/master/utils/docker-compose/Makefile), etc. If you search for "python3-package.mk" in the repo you will find many other examples. I suggest you take one of these makefiles as a template.
> 
> The Python packages folder readme also has a section on [building a Python package](https://github.com/openwrt/packages/blob/master/lang/python/README.md#building-a-python-package) which explains most (if not all) of the various Python package variables.
> 
> If you run into any problems writing a Python package makefile, there are lots of folks here who can help, but I'm afraid your `install_suricata_update` is definitely not the way to do this.

I absolutely agree.  What I have been playing with is simply a file check on the suricata-update command as part of the init.d service and calling `pip` to install it if it isn't there.  This allows me to include the service and have it self-installing without a scripted mess.  I've been busy with other things, but yes, this was totally a cluster 👍 

I'll get a push up eventually and see if that isn't more acceptable.  Until I get rust in, it's pretty much a moot point though.  It's still all a WIP.

**Comment by BKPepe** [comment]
Closing, OP is inactive.

**Pull request closed without merging**